--- a/kotlin
+++ b/kotlin
@@ -17,9 +17,9 @@
 set -e -u
 
 # The version of the Kotlin Toolchain (and CLI) distribution to provision and use
-kotlin_cli_version=0.11.1
+kotlin_cli_version=0.12.2
 # Establish chain of trust from here by specifying exact checksum of Kotlin Toolchain (and CLI) distribution to be run
-kotlin_cli_sha256=0ded2a434f6bf193b24e2a6d56c3ba443f4232721155a65aaa8372789412112f
+kotlin_cli_sha256=9fec42de4f378a27240b78c1b66ffef0c2aca66a4a6390ca67fe51aacb75677e
 
 KOTLIN_CLI_DOWNLOAD_ROOT="${KOTLIN_CLI_DOWNLOAD_ROOT:-https://packages.jetbrains.team/maven/p/amper/amper}"
 
--- a/kotlin.bat
+++ b/kotlin.bat
@@ -17,9 +17,9 @@
 setlocal
 
 @rem The version of the Kotlin Toolchain distribution to provision and use
-set kotlin_cli_version=0.11.1
+set kotlin_cli_version=0.12.2
 @rem Establish chain of trust from here by specifying the exact checksum of the Kotlin Toolchain distribution to be run
-set kotlin_cli_sha256=0ded2a434f6bf193b24e2a6d56c3ba443f4232721155a65aaa8372789412112f
+set kotlin_cli_sha256=9fec42de4f378a27240b78c1b66ffef0c2aca66a4a6390ca67fe51aacb75677e
 
 if not defined KOTLIN_CLI_DOWNLOAD_ROOT set KOTLIN_CLI_DOWNLOAD_ROOT=https://packages.jetbrains.team/maven/p/amper/amper
 if not defined KOTLIN_CLI_BOOTSTRAP_CACHE_DIR set KOTLIN_CLI_BOOTSTRAP_CACHE_DIR=%LOCALAPPDATA%\JetBrains\Kotlin\cli
@@ -175,7 +175,7 @@
 
 if exist "%project_dir%\project.yaml" (
     @rem Found project.yaml but no wrapper alongside it
-    echo WARNING: Found a project.yaml in '%project_dir%', but the wrapper script is missing; using Kotlin Toolchain v$kotlin_cli_version. >&2
+    echo WARNING: Found a project.yaml in '%project_dir%', but the wrapper script is missing; using Kotlin Toolchain v%kotlin_cli_version%. >&2
     exit /b 1
 )
 
@@ -253,5 +253,12 @@
 rem We use busybox here because it doesn't reinterpret the user-passed command-line arguments (that we pass via %*).
 rem Also this way we can use the unified launcher script (.sh)
 set KOTLIN_CLI_WRAPPER_PATH=%~f0
-"%busybox_exe%" sh "%kotlin_cli_target_dir%\bin\launcher.sh" %*
-exit /B %ERRORLEVEL%
+
+rem The '& call' after the app run is to avoid the "Terminate batch job (Y/N)?" prompt
+"%busybox_exe%" sh "%kotlin_cli_target_dir%\bin\launcher.sh" %* & call :exitWithErrorLevel
+
+:exitWithErrorLevel
+@rem We use "%COMSPEC%" /d /c exit so that and/or operators work properly when calling kotlin.bat directly from a script
+@rem without the `call` command. With a plain `exit /B %ERRORLEVEL%`, using `kotlin.bat && echo success` would print
+@rem 'success' even in case of failure. Consumers would have to use `call kotlin.bat && echo success` for it to work.`
+"%COMSPEC%" /d /c exit %ERRORLEVEL%
--- a/shared-core/module.yaml
+++ b/shared-core/module.yaml
@@ -1,6 +1,6 @@
 product:
   type: kmp/lib
-  platforms: [android, iosArm64, iosSimulatorArm64, iosX64]
+  platforms: [android, iosArm64, iosSimulatorArm64]
 
 dependencies:
   - dev.zacsweers.metro:runtime:1.1.1
--- a/shared-di/module.yaml
+++ b/shared-di/module.yaml
@@ -1,6 +1,6 @@
 product:
   type: kmp/lib
-  platforms: [android, iosArm64, iosSimulatorArm64, iosX64]
+  platforms: [android, iosArm64, iosSimulatorArm64]
 
 dependencies:
   - ../shared-core
--- a/shared-feature-home/module.yaml
+++ b/shared-feature-home/module.yaml
@@ -1,6 +1,6 @@
 product:
   type: kmp/lib
-  platforms: [android, iosArm64, iosSimulatorArm64, iosX64]
+  platforms: [android, iosArm64, iosSimulatorArm64]
 
 dependencies:
   - ../shared-core
--- a/shared-feature-nearby-vehicle-map/module.yaml
+++ b/shared-feature-nearby-vehicle-map/module.yaml
@@ -1,6 +1,6 @@
 product:
   type: kmp/lib
-  platforms: [android, iosArm64, iosSimulatorArm64, iosX64]
+  platforms: [android, iosArm64, iosSimulatorArm64]
 
 dependencies:
   - ../shared-core
--- a/shared-ui-home/module.yaml
+++ b/shared-ui-home/module.yaml
@@ -1,6 +1,6 @@
 product:
   type: kmp/lib
-  platforms: [android, iosArm64, iosSimulatorArm64, iosX64]
+  platforms: [android, iosArm64, iosSimulatorArm64]
 
 dependencies:
   - ../shared-feature-home
--- a/gradle-bridge/shared-kit/build.gradle.kts
+++ b/gradle-bridge/shared-kit/build.gradle.kts
@@ -18,7 +18,6 @@
 }
 
 kotlin {
-    iosX64()
     iosArm64()
     iosSimulatorArm64()
 
@@ -61,13 +60,6 @@
             )
         }
 
-        val iosX64Main by getting {
-            kotlin.srcDirs(
-                "../../shared-core/src@ios",
-                "../../shared-ui-home/src@ios",
-            )
-        }
-
         val iosArm64Main by getting {
             kotlin.srcDirs(
                 "../../shared-core/src@ios",
